src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -380,5 +381,34 @@ public void testCloseConnectionOnMakeObject() {
   private int getClientCount(final String clientList) {
     return clientList.split("\n").length;
   }
-
+  
+  @Test(expected = JedisConnectionException.class)
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000, "foobared", 0, "my_shiny_client_name", false, null, null, null);
+    JedisPool pool = new JedisPool(new JedisPoolConfig(), factory);
+    Jedis obj1 = pool.getResource();
+    obj1.set("foo", "bar");
+    assertEquals("bar", obj1.get("foo"));
+    assertEquals(1, pool.getNumActive());
+    
+    factory.setPassword("wrong password");
```

Comment:
Let's wrap with try-catch and remove expected statement. The test will pass even JedisConnectionException is thrown from obj1.

Reply:
done